spring-integration-core/src/main/java/org/springframework/integration/support/converter/MapMessageConverter.java

```diff
@@ -77,12 +78,16 @@ protected MessageBuilderFactory getMessageBuilderFactory() {
 	 * @param headerNames The header names.
 	 */
 	public void setHeaderNames(String... headerNames) {
-		Assert.notEmpty(headerNames, "at least one header name is required");
-		this.headerNames = Arrays.copyOf(headerNames, headerNames.length);
+		if (ObjectUtils.isEmpty(headerNames)) {
+			this.headerNames = new String[] {};
```

Comment:
Why? With `Assert.notEmpty()` we just disallow to set a null into this property.
I don't see anything what really has to be fixed in this method.
The default one is empty array, but if it is set, it must not be empty.
That's it. What wrong do you see with that logic?

Reply:
With the current implementation, it can be changed to anything but the default state during runtime. So setting back to an empty array is not possible.

Reply:
OK. That makes sense.
I'll pull your PR locally shortly for final review and possible merge.

Thank you!